rust/src/dataset.rs

```diff
@@ -147,6 +140,13 @@ impl Dataset {
                         // offsets
                         let cur_src_pos = cur_box.min() - dst_pos;
                         let cur_dst_pos = cur_box.min() - cur_file_box.min();
+                        if self.header.block_type != BlockType::Raw {  
+                            if cur_box != cur_file_box {
```

Comment:
If I try to write a (1025, 1024, 1024) data volume to location (0, 0, 0), this will write the first (1024)³ cube and subsequently throw an error for the (1, 1024, 1024) plane at (1024, 0, 0). This partial write is problematic because it leaves the dataset in an inconsistent state.

Please check that `dst_pos` and `mat.shape` are zero modulo the file volume at the start of the function.